app/routers/google_connect.py

```diff
@@ -0,0 +1,199 @@
+from app.routers.profile import router as profile_router
+from app.config import CLIENT_SECRET_FILE
+from app.routers.event import create_event
+
+SCOPES = ['https://www.googleapis.com/auth/calendar']
+
+router = APIRouter(
+    prefix="/google",
+    tags=["sync"],
+    responses={404: {"description": "Not found"}},
+)
+
+
+@router.get("/sync")
+async def google_sync(session=Depends(get_db)) -> RedirectResponse:
+    '''Sync with Google - if user never synced with google this funcion will take
+    the user to a consent screen to use his google calendar data with the app.
+    '''
+
+    # TODO - get connected/current user
+    user = session.query(User).filter_by(id=1).first()
+    credentials, status = get_credentials_from_db(user)
+    if status:
+        if credentials.expired:
+            credentials = refresh_token(credentials, session, user)
+
+    elif not status:
+        # first sync
+        if isClientSecretNotNone():  # if there is no client_secrets.json
+            print('Google Sync is not available - missing client_secret.json')
+            url = profile_router.url_path_for("profile")
+            return RedirectResponse(url=url)
+
+        flow = InstalledAppFlow.from_client_secrets_file(
+            client_secrets_file=CLIENT_SECRET_FILE,
+            scopes=SCOPES
+        )
+
+        flow.run_local_server(prompt='consent')
+        credentials = flow.credentials
+
+        oauth_credentials = OAuthCredentials(
+            owner=user,
+            token=credentials.token,
+            refresh_token=credentials.refresh_token,
+            token_uri=credentials.token_uri,
+            client_id=credentials.client_id,
+            client_secret=credentials.client_secret,
+            expiry=credentials.expiry
+        )
+
+        session.add(oauth_credentials)
+        session.commit()
+
+    print("clean up db")
+    session.query(OAuthCredentials).filter_by(user_id=None).delete()
+    session.commit()
+
+    events = get_current_year_events(credentials, user, session)
+    push_events_to_db(events, user, session)
+    session.close()
+
+    url = profile_router.url_path_for("profile")
+    return RedirectResponse(url=url)
+
+
+def isClientSecretNotNone():
+    return CLIENT_SECRET_FILE is None
+
+
+def get_current_year_events(
+                credentials: Credentials, user: User, session: SessionLocal):
+    '''Getting user events from google calendar'''
+
+    currrnt_year = datetime.now().year
+    start = datetime(currrnt_year, 1, 1).isoformat() + 'Z'
+    end = datetime(currrnt_year + 1, 1, 1).isoformat() + 'Z'
+
+    service = build('calendar', 'v3', credentials=credentials)
+    events_result = service.events().list(
+        calendarId='primary',
+        timeMin=start,
+        timeMax=end,
+        singleEvents=True,
+        orderBy='startTime'
+    ).execute()
+
+    events = events_result.get('items', [])
+    return events
+
+
+def push_events_to_db(events: list, user: User, session: SessionLocal) -> bool:
```

Comment:
This function is really tough to read as it doesn't enters in a single screen.
Can think of ways to shorten it?

Reply:
I cant shorten it because it's doing one thing, but i can add some comments :)